op-service/sources/types.go

```diff
@@ -225,9 +225,10 @@ func (block *RPCBlock) verify() error {
 			return fmt.Errorf("block tx %d is nil", i)
 		}
 	}
-	if computed := types.DeriveSha(types.Transactions(block.Transactions), trie.NewStackTrie(nil)); block.TxHash != computed {
-		return fmt.Errorf("failed to verify transactions list: computed %s but RPC said %s", computed, block.TxHash)
-	}
+	// TODO reinstate
+	// if computed := types.DeriveSha(block.Transactions, trie.NewStackTrie(nil)); block.TxHash != computed {
+	// 	return fmt.Errorf("failed to verify transactions list: computed %s but RPC said %s", computed, block.TxHash)
+	// }
```

Comment:
This is a bit of a fundamental roadblock. There is not a way for us compute the binary encoding of a future / unsupported transaction which we receive over JSON RPC. So we wouldn't be able to trustlessly verify a block containing such transactions.

On the op-program side it can still work since it receives all transactions in their binary encoding anyway (from the PreImageOracle). So in this case we can make a new type `GenericTxs = []GenericTx` with the requisite `Len()` and `EncodeIndex(int, *bytes.Buffer)` method (so that it implements op-geth's  `DerivableList` types.

Reply:
See discord discussion and https://github.com/ethereum-optimism/op-geth/pull/468/files